HIP/hip-1340.md

```diff
@@ -0,0 +1,388 @@
+
+This will be accomplished via the creation of a new `ContractServiceApi` whose interface will be defined as follows:
+
+```
+/**
+ * Defines mutations that can't be expressed as a {@link com.hedera.hapi.node.transaction.TransactionBody} dispatch.
+ *
+ * <p>Only exported to the token service at this time, as it is the only service that currently needs such a thing.
+ */
+public interface ContractServiceApi {
+    /**
+     * Sets the given address as a delegation target of the specified account.
+     * The address will be prefixed with `0xef0100` and stored as runtime bytecode of the given account.
+     * Setting the target address to `0x0000000000000000000000000000000000000000` results in clearing
+     * the Delegation Indicator of the given account (i.e. setting its runtime bytecode to an empty array).
+     *
+     * @param accountID         id of the account to set the delegation for
+     * @param delegationTarget  the delegation target address
+     */
+    void setAccountDelegationTarget(@NonNull AccountID accountID, @NonNull Address delegationTarget);
+}
+```
+
+##### Updating Code Delegation Via a New EVM transaction Type 4, As Defined in EIP-7702
+
+EIP-7702 defines a new EVM transaction type, which extends the previous transaction format with an additional `authorization_list` field:
+ ```
+    transaction_payload = rlp([chain_id, nonce, max_priority_fee_per_gas, max_fee_per_gas, gas_limit,
+                              destination, value, data, access_list, authorization_list, signature_y_parity,
+                              signature_r, signature_s])
+
+    authorization_list = [[chain_id, address, nonce, y_parity, r, s], ...]
+```
+
+Each entry on the authorization list is an independent intent signifying that the signer of a specific message (consisting of chain_id, address and nonce, defined in detail in EIP-7702) wishes to update their EOA delegation indicator to `address`.
+The signer of the message is derived from the components of the signature (`r`, `s`, `y_parity`) and the signed payload using `ecrecover`.
+Simplifying, for each item on the list do: `account = ecrecover(r, s, msg); account.setCode(0xef0100 || address)`.
+
+The validity of each item will be checked closely following EIP-7702's specification, which includes (but is not limited to) verifying the correctness of the signature, chain ID and nonce.
+
+If more than one valid authorization entry for the same EOA exists on the list, then only the last one will be used. Any authorizations that come before it will be ignored.
+
+Each successfully processed authorization item will result in a child `CryptoCreate` or `CryptoUpdate` dispatch (depending on whether the affected EOA already exists or not), carrying the intended Delegation Indicator change. No child dispatch will be created for duplicate authorizations that were skipped.
```

Comment:
Will these be preceding synthetic creates or after the parent? I assume since there's no transfers to them then it should be okay to be after.